arviz/utils.py

```diff
@@ -601,3 +601,25 @@ def get_coords(data, coords):
         except KeyError as err:
             raise KeyError("Error in data[{}]: {}".format(idx, err))
     return data_subset
+
+
+def credible_interval_warning(credible_interval, hpd_interval):
+    """Replace credible_interval with hpd_interval and to warns of be deprecation."""
+    warnings.warn(
+        (
+            "Keyword argument credible_interval has been deprecated "
+            "Please replace with hpd_interval"
+        ),
+    )
+
+    if isinstance(credible_interval, str) and credible_interval == "auto":
+        raise Exception("Argument value 'auto' has been renamed to 'hide'")
```

Comment:
the exception should be different. people who was using `"auto"` (probably nobody) now have to use `None`, its people using `None` whose plots will change and should modify the code to use `"hide"`. I don't see any way to warn people who used to pass `None` other than release notes. Maybe in the deprecation warning? it is only relevant for plot_posterior though.

Reply:
Ohhhhhhhhhhhhhhhhhh that makes sense. I 100% agree with you. I will add a warning

Reply:
Ok now thinking about it I'm stuck. because if I check for `None` then a warning will be raised by default since that's now the default arg, which I think will happen for most users. but if I dont then the users who were specifying `None ` explicitly will get confused, but I think that's a smaller subset.

Do you think the docs will help if people get confused without needing to make it as clear during runtime?

Reply:
True, only alternative that comes to mind is setting the default of `credible_interval` to `"deprecated"` or some other new and unrelated value to clearly distinguish when are users actually passing a value to it, not sure if it is a good idea or if it will be even more confusing though.